Return-path: <linux-media-owner@vger.kernel.org>
Received: from cantor2.suse.de ([195.135.220.15]:50792 "EHLO mx2.suse.de"
	rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
	id S932339AbbBIK7K (ORCPT <rfc822;linux-media@vger.kernel.org>);
	Mon, 9 Feb 2015 05:59:10 -0500
Date: Mon, 09 Feb 2015 11:59:07 +0100
Message-ID: <s5hvbjbtkp0.wl-tiwai@suse.de>
From: Takashi Iwai <tiwai@suse.de>
To: Mauro Carvalho Chehab <mchehab@osg.samsung.com>
Cc: Hans Verkuil <hverkuil@xs4all.nl>, linux-media@vger.kernel.org,
	linux-kernel@vger.kernel.org
Subject: DVB suspend/resume regression on 3.19
MIME-Version: 1.0 (generated by SEMI 1.14.6 - "Maruoka")
Content-Type: text/plain; charset=US-ASCII
Sender: linux-media-owner@vger.kernel.org
List-ID: <linux-media.vger.kernel.org>

Hi,

we've got a bug report about the suspend/resume regression of DVB
device with 3.19.  The symptom is VLC doesn't work after S3 or S4
resume.  strace shows that /dev/dvb/adaptor0/dvr returns -ENODEV.

The reporter confirmed that 3.18 works fine, so the regression must be
in 3.19.

There is a relevant kernel warning while suspending:

 WARNING: CPU: 1 PID: 3603 at ../kernel/module.c:1001 module_put+0xc7/0xd0()
 Workqueue: events_unbound async_run_entry_fn
  0000000000000000 ffffffff81a45779 ffffffff81664f12 0000000000000000
  ffffffff81062381 0000000000000000 ffffffffa051eea0 ffff8800ca369278
  ffffffffa051a068 ffff8800c0a18090 ffffffff810dfb47 0000000000000000
 Call Trace:
  [<ffffffff810055ac>] dump_trace+0x8c/0x340
  [<ffffffff81005903>] show_stack_log_lvl+0xa3/0x190
  [<ffffffff81007061>] show_stack+0x21/0x50
  [<ffffffff81664f12>] dump_stack+0x47/0x67
  [<ffffffff81062381>] warn_slowpath_common+0x81/0xb0
  [<ffffffff810dfb47>] module_put+0xc7/0xd0
  [<ffffffffa04d98d1>] dvb_usb_adapter_frontend_exit+0x41/0x60 [dvb_usb]
  [<ffffffffa04d8451>] dvb_usb_exit+0x31/0xa0 [dvb_usb]
  [<ffffffffa04d84fb>] dvb_usb_device_exit+0x3b/0x50 [dvb_usb]
  [<ffffffff814cefad>] usb_unbind_interface+0x1ed/0x2c0
  [<ffffffff8145ceae>] __device_release_driver+0x7e/0x100
  [<ffffffff8145cf52>] device_release_driver+0x22/0x30
  [<ffffffff814cf13d>] usb_forced_unbind_intf+0x2d/0x60
  [<ffffffff814cf3c3>] usb_suspend+0x73/0x130
  [<ffffffff814bd453>] usb_dev_freeze+0x13/0x20
  [<ffffffff81468fca>] dpm_run_callback+0x4a/0x150
  [<ffffffff81469c81>] __device_suspend+0x121/0x350
  [<ffffffff81469ece>] async_suspend+0x1e/0xa0
  [<ffffffff81081e63>] async_run_entry_fn+0x43/0x150
  [<ffffffff81079e72>] process_one_work+0x142/0x3f0
  [<ffffffff8107a234>] worker_thread+0x114/0x460
  [<ffffffff8107f3b1>] kthread+0xc1/0xe0
  [<ffffffff8166b77c>] ret_from_fork+0x7c/0xb0

So something went wrong in module refcount, which likely leads to
disabling the device and returning -ENODEV in the end.

Does this ring a bell to you guys?

The hardware details and logs are found in the URL below:
  https://bugzilla.novell.com/show_bug.cgi?id=916577


thanks,

Takashi
